coreos-koji-tagger/coreos_koji_tagger.py

```diff
@@ -447,22 +447,22 @@ def process_lockfiles(self, rev):
                 watch_tasks(self.koji_client, [task], poll_interval=10)
                 logger.info('Dist-repo task has finished')
 
-    # retry to login every 30s for 15m before giving up
-    # https://github.com/coreos/fedora-coreos-releng-automation/issues/70
-    @retry(retry=retry_if_exception_type(koji.AuthError), wait=wait_fixed(30), stop=stop_after_attempt(30))
     def koji_login(self):
-        # If already authenticated then nothing to do
-        # Catch koji.AuthError as that is what happens
-        # when we get logged out.
+        principal = find_principal_from_keytab(self.keytab_file)
+        self.koji_client.gssapi_login(principal, self.keytab_file)
+
+    # Check the login. If it's bad then we'll exit the pod so a new one will get created.
+    # This is the only way we've come up with to deal with the problem in
+    # https://github.com/coreos/fedora-coreos-releng-automation/issues/70
+    def koji_check_login(self):
+        # If currently logged in then this will pass and koji.AuthError won't get thrown
         try:
             if self.koji_client.getLoggedInUser():
                 return
         except koji.AuthError as e:
-            logger.info('Received koji.AuthError from koji. Re-attempting login.')
-            pass
-        # Login!
-        principal = find_principal_from_keytab(self.keytab_file)
-        self.koji_client.gssapi_login(principal, self.keytab_file)
+            logger.error('Received koji.AuthError from koji. Exiting so a new pod will get created.')
+            logger.error('See https://github.com/coreos/fedora-coreos-releng-automation/issues/70')
+            sys.exit(1)
```

Comment:
Rather than doing this here, why not let `koji.AuthError` propagate, and special-case it in `catch_exceptions_and_continue`?

Reply:
just pushed up a new commit, is that what you mean? If so I'll squash the commit into the previous commit.